vck/src/commonMain/kotlin/at/asitplus/wallet/lib/jws/JwsService.kt

```diff
@@ -272,92 +286,102 @@ object DecryptJwe {
 
 }
 
-interface VerifierJwsService {
-
-    val supportedAlgorithms: List<JwsAlgorithm>
-
-    suspend fun verifyJwsObject(jwsObject: JwsSigned<*>): Boolean
-
-    suspend fun verifyJws(jwsObject: JwsSigned<*>, signer: JsonWebKey): Boolean
-
-    suspend fun verifyJws(jwsObject: JwsSigned<*>, cnf: ConfirmationClaim): Boolean
-
-}
 
 /**
  * Clients need to retrieve the URL passed in as the only argument, and parse the content to [JsonWebKeySet].
  */
-typealias JwkSetRetrieverFunction = suspend (String) -> JsonWebKeySet?
+fun interface JwkSetRetrieverFunction {
+    operator fun invoke(url: String): JsonWebKeySet?
+}
 
 /**
  * Clients get the parsed [JwsSigned] and need to provide a set of keys, which will be used for verification one-by-one.
  */
-typealias PublicJsonWebKeyLookup = suspend (JwsSigned<*>) -> Set<JsonWebKey>?
-
-typealias VerifyJwsSignatureFun = suspend (jwsObject: JwsSigned<*>, publicKey: CryptoPublicKey) -> Boolean
+fun interface PublicJsonWebKeyLookup {
+    suspend operator fun invoke(
+        jwsObject: JwsSigned<*>,
+    ): Set<JsonWebKey>?
+}
 
-object VerifyJwsSignature {
+fun interface VerifyJwsSignatureFun {
     operator fun invoke(
-        verifySignature: VerifySignatureFun = VerifySignature(),
-    ): VerifyJwsSignatureFun = { jwsObject, publicKey ->
-        catching {
-            verifySignature(
-                jwsObject.plainSignatureInput,
-                jwsObject.signature,
-                jwsObject.header.algorithm.algorithm,
-                publicKey,
-            ).getOrThrow()
-        }.fold(
-            onSuccess = { true },
-            onFailure = {
-                Napier.w("No verification from native code", it)
-                false
-            })
-    }
+        jwsObject: JwsSigned<*>,
+        publicKey: CryptoPublicKey,
+    ): Boolean
 }
 
-typealias VerifyJwsSignatureWithKeyFun = suspend (jwsObject: JwsSigned<*>, signer: JsonWebKey) -> Boolean
+class VerifyJwsSignature(
+    val verifySignature: VerifySignatureFun = VerifySignature(),
+) : VerifyJwsSignatureFun {
+    override operator fun invoke(
```

Comment:
Considering we will be adding remote PQC capabilities to Signum, this should be suspending